admin/resources/js/components/H2Components/Select.tsx

```diff
@@ -0,0 +1,123 @@
+  label: string;
+  /** A string specifying a name for the input control. */
+  name: string;
+  /** List of options for the select element. */
+  options: { value: string | number; text: string }[];
+  /** Object set of validation rules to impose on input. */
+  rules?: RegisterOptions;
+  /** Optional context which user can view by toggling a button. */
+  context?: string;
+  /** If an error string is provided, it will appear below the fieldset inputs. */
+  error?: string;
+}
+
+const Select: React.FunctionComponent<SelectProps> = ({
+  id,
+  label,
+  name,
+  options,
+  rules,
+  context,
+  error,
+  ...rest
+}) => {
+  const [contextIsActive, setContextIsActive] = useState(false);
+
+  const {
+    register,
+    formState: { errors },
+  } = useFormContext();
+  const formError = errors[name]?.message;
+  return (
+    <div>
+      <div data-h2-flex-grid="b(top, contained, flush, s)">
+        <div data-h2-flex-item="b(1of2)">
+          <label data-h2-font-size="b(caption)" data-h2-bg-color="b(theme-1)">
+            {label}
+          </label>
+        </div>
+        <div data-h2-flex-item="b(1of2)">
+          {context && (
+            <button
+              type="button"
+              style={{ float: "right" }}
+              className="input-label-context-button"
+              data-h2-margin="b(left, xxs)"
+              title="Toggle Context"
+              onClick={() =>
+                setContextIsActive((currentState) => !currentState)
+              }
+            >
+              <>
+                {contextIsActive ? (
+                  <XCircleIcon
+                    style={{ width: "calc(1rem/1.25)" }}
+                    data-h2-font-color="b(lightpurple)"
+                  />
+                ) : (
+                  <QuestionMarkCircleIcon
+                    style={{ width: "calc(1rem/1.25)" }}
+                    data-h2-font-color="b(lightpurple)"
+                  />
+                )}
+              </>
+            </button>
+          )}
+          <label
+            data-h2-font-size="b(caption)"
+            data-h2-bg-color="b(theme-1)"
+            data-h2-padding="b(left, xxs)"
+            style={{ float: "right" }}
+          >
+            {rules && rules?.required && (
+              <span data-h2-color="b(red)">Required</span>
+            )}
+            {!rules?.required && <span>Optional</span>}
+          </label>
+        </div>
+      </div>
+      <select
+        data-h2-radius="b(s)"
+        data-h2-padding="b(top-bottom, xxs) b(left, xxs)"
+        data-h2-margin="b(bottom, xs)"
+        data-h2-font-size="b(normal)"
+        id={id}
+        style={{ width: "100%" }}
+        {...register(name, rules)}
+        aria-invalid={error ? "true" : "false"}
+        {...rest}
+      >
+        {options.map(({ value, text }) => (
+          <option
+            data-h2-font-size="b(caption)"
+            data-h2-color="b(white)"
```

Comment:
Did you mean to use data-h2-bg-color (for background color)? The current white text is unreadable.

Reply:
Aye, this was a mistake. This actually changed nothing on my screen so I forgot about it. Will remove this.